core/src/main/scala/kafka/zk/ZkData.scala

```diff
@@ -141,15 +139,33 @@ object ConfigEntityZNode {
     Json.encodeAsBytes(Map("version" -> 1, "config" -> config.asScala))
   }
   def decode(bytes: Array[Byte]): Option[Properties] = {
-    Json.parseBytes(bytes).map { js =>
-      val configOpt = js.asJsonObjectOption.flatMap(_.get("config").flatMap(_.asJsonObjectOption))
-      val props = new Properties()
-      configOpt.foreach(config => config.iterator.foreach { case (k, v) => props.setProperty(k, v.to[String]) })
-      props
+    val props = new Properties()
+    if (bytes != null) {
+      Json.parseBytes(bytes).map { js =>
+        val jsObject = js.asJsonObjectOption.getOrElse {
+          throw new IllegalArgumentException(s"Unexpected value in config:")
+        }
+        require(jsObject("version").to[Int] == 1)
```

Comment:
The decode() method is currently called by the admin tools. The issue is that if we ever evolve the version field in the future, admin tools on the old version will fail. So, we probably want to relax the check.

Reply:
@junrao  Thanks for the review. Updated the PR with review comments.